proto/cosmwasm/wasm/v1/tx.proto

```diff
@@ -64,6 +69,31 @@ message MsgInstantiateContract {
     (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
   ];
 }
+
+// MsgInstantiateContract2 create a new smart contract instance for the given
+// code id with a predicable address.
+message MsgInstantiateContract2 {
+  // Sender is the that actor that signed the messages
+  string sender = 1;
+  // Admin is an optional address that can execute migrations
+  string admin = 2;
+  // CodeID is the reference to the stored WASM code
+  uint64 code_id = 3 [ (gogoproto.customname) = "CodeID" ];
+  // Label is optional metadata to be stored with a contract instance.
+  string label = 4;
+  // Msg json encoded message to be passed to the contract on instantiation
+  bytes msg = 5 [ (gogoproto.casttype) = "RawContractMessage" ];
+  // Funds coins that are transferred to the contract on instantiation
+  repeated cosmos.base.v1beta1.Coin funds = 6 [
+    (gogoproto.nullable) = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+  // Salt is an arbitrary value provided by the sender
+  bytes salt = 7;
+  // IncludeInitMsg into the hash for the predictable address. Default is false
+  bool include_init_msg = 8;
```

Comment:
"init" was removed from all places as far as I can tell. It's no good abbreviation for "instantiate". Also in the context of MsgInstantiateContract2, the message is always included. We need something that makes clear this is for the address generation.

I suggest `fix_msg` which means we fix the message as part of the address generation.